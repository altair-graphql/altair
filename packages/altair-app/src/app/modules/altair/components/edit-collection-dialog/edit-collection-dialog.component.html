<nz-modal
  [nzVisible]="showEditCollectionDialog"
  (nzVisibleChange)="toggleDialogChange.next($event)"
  [nzTitle]="modalTitle"
  [nzContent]="modalContent"
  [nzFooter]="modalFooter"
  [nzWidth]="800"
  (nzOnCancel)="toggleDialogChange.next($event)"
>
  <ng-template #modalTitle>
    <div class="app-dialog-header">
      <div class="app-dialog-title">{{ 'EDIT_COLLECTION_TITLE_TEXT' | translate }}</div>
      {{ 'EDIT_COLLECTION_TITLE_SUBTEXT' | translate }}
    </div>
  </ng-template>

  <ng-template #modalContent>
    <div class="app-dialog-body" *ngIf="collection">
      <div class="app-dialog-section">
        <div class="form-group">
          {{ 'COLLECTION_TITLE_LABEL' | translate }}
          <input type="text" class="dialog-input" [(ngModel)]="title">
        </div>
      </div>
      <div class="app-dialog-section">
        <nz-tabset
          [nzAnimated]="false"
          nzSize="small"
          nzTabBarGutter="16"
        >
          <nz-tab [nzTitle]="'PRE_REQUEST_TAB' | translate">
            <app-pre-request-editor
              [preRequest]="preRequest"
              (preRequestScriptChange)="preRequest.script = $event"
              (preRequestEnabledChange)="preRequest.enabled = $event"
            ></app-pre-request-editor>
          </nz-tab>
          <nz-tab [nzTitle]="'POST_REQUEST_TAB' | translate">
            <app-post-request-editor
              [postRequest]="postRequest"
              (postRequestScriptChange)="postRequest.script = $event"
              (postRequestEnabledChange)="postRequest.enabled = $event"
            ></app-post-request-editor>
          </nz-tab>
        </nz-tabset>
      </div>
    </div>
  </ng-template>

  <ng-template #modalFooter>
    <div class="app-dialog-footer">
      <button class="app-button active-primary right" track-id="" (click)="updateCollection()">{{ 'SAVE_BUTTON' | translate }}</button>
    </div>
  </ng-template>
</nz-modal>
