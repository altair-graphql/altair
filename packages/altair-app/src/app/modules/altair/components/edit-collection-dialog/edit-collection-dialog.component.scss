.headers-section {
  &__header {
    display: flex;
    justify-content: space-between;
    align-items: center;
    margin-bottom: 20px;
    
    span {
      font-size: 14px;
      color: var(--theme-font-color);
    }
  }
}

.set-header-input-container {
  .header-input {
    flex: 1;
    margin: 0 5px;
    
    .input {
      width: 100%;
    }
  }
  
  .header-input-remove-button {
    background: none;
    border: none;
    color: var(--theme-font-color);
    font-size: 18px;
    cursor: pointer;
    margin-left: 10px;
    opacity: 0.7;
    
    &:hover {
      opacity: 1;
      color: var(--danger-color);
    }
  }
}

.no-headers-message {
  text-align: center;
  color: var(--theme-font-color);
  opacity: 0.7;
  font-style: italic;
  padding: 20px;
}

.headers__notice {
  margin-top: 16px;
  padding: 12px;
  background: rgba(var(--rgb-primary-color), 0.1);
  border-radius: 4px;
  font-size: 12px;
  color: var(--theme-font-color);
  opacity: 0.8;
  
  a {
    color: var(--primary-color);
    text-decoration: none;
    
    &:hover {
      text-decoration: underline;
    }
  }
}
