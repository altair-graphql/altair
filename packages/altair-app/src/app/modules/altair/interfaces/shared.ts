export interface IDictionary<V = any> {
  [key: string]: V;
}

export type Omit<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>>;

export type TODO = any;

export type UnknownError = TODO;

export interface TrackByIdItem {
  id: string;
}
