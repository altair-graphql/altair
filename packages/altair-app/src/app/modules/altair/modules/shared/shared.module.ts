import { NgModule, ModuleWithProviders } from '@angular/core';
import { CommonModule } from '@angular/common';

import { DragDropModule } from '@angular/cdk/drag-drop';
import { TranslateModule } from '@ngx-translate/core';
import { NgPipesModule } from 'ngx-pipes';
import { NguiAutoCompleteModule } from '@ngui/auto-complete';
import { MarkdownModule } from 'ngx-markdown';
import { HotToastModule } from '@ngneat/hot-toast';
import { ResizableModule } from 'angular-resizable-element';

import { NzButtonModule } from 'ng-zorro-antd/button';
import { NzFormModule } from 'ng-zorro-antd/form';
import { NzSelectModule } from 'ng-zorro-antd/select';
import { NzModalModule } from 'ng-zorro-antd/modal';
import { NzDropDownModule } from 'ng-zorro-antd/dropdown';
import { NzSwitchModule } from 'ng-zorro-antd/switch';
import { NzListModule } from 'ng-zorro-antd/list';
import { NzRadioModule } from 'ng-zorro-antd/radio';
import { NzAlertModule } from 'ng-zorro-antd/alert';
import { NzLayoutModule } from 'ng-zorro-antd/layout';
import { NzMenuModule } from 'ng-zorro-antd/menu';
import { NzCollapseModule } from 'ng-zorro-antd/collapse';
import { NzInputModule } from 'ng-zorro-antd/input';
import { NzInputNumberModule } from 'ng-zorro-antd/input-number';
import { NzAutocompleteModule } from 'ng-zorro-antd/auto-complete';
import { NzCheckboxModule } from 'ng-zorro-antd/checkbox';
import { NzToolTipModule } from 'ng-zorro-antd/tooltip';
import { NzTabsModule } from 'ng-zorro-antd/tabs';
import { NzTableModule } from 'ng-zorro-antd/table';
import { NzTreeSelectModule } from 'ng-zorro-antd/tree-select';
import { NzGridModule } from 'ng-zorro-antd/grid';
import { NzPopoverModule } from 'ng-zorro-antd/popover';

import { IconsModule } from '../icons/icons.module';
import { NzAvatarModule } from 'ng-zorro-antd/avatar';

const AntdComponentModules = [
  NzButtonModule,
  NzInputModule,
  NzInputNumberModule,
  NzAutocompleteModule,
  NzCheckboxModule,
  NzFormModule,
  NzGridModule,
  NzTabsModule,
  NzSelectModule,
  NzModalModule,
  NzDropDownModule,
  NzSwitchModule,
  NzListModule,
  NzRadioModule,
  NzAlertModule,
  NzLayoutModule,
  NzMenuModule,
  NzCollapseModule,
  NzToolTipModule,
  NzTableModule,
  NzTreeSelectModule,
  NzPopoverModule,
  NzAvatarModule,
  // NgZorroAntdModule,
];

@NgModule({
  imports: [
    CommonModule,
    DragDropModule,
    TranslateModule.forRoot(),
    NgPipesModule,
    NguiAutoCompleteModule, // use antd
    MarkdownModule.forRoot(),
    ResizableModule,
    IconsModule,
    // ToastrModule.forRoot({
    //   newestOnTop: false,
    //   closeButton: true,
    //   positionClass: 'toast-top-center',
    //   enableHtml: true,
    //   countDuplicates: true,
    //   preventDuplicates: true,
    //   resetTimeoutOnDuplicate: true,
    // }),
    // HotToastModule.forRoot({
    //   position: 'top-center',
    //   reverseOrder: true,
    //   dismissible: true,
    //   closeStyle: {
    //     backgroundImage: `none`,
    //   },
    // }),
    ...AntdComponentModules,
  ],

  // DON'T FORGET TO EXPORT MODULE
  exports: [
    DragDropModule,
    TranslateModule,
    NgPipesModule,
    NguiAutoCompleteModule,
    MarkdownModule,
    ResizableModule,
    IconsModule,
    HotToastModule,
    ...AntdComponentModules,
  ],
})
export class SharedModule {
  static forRoot(): ModuleWithProviders<SharedModule> {
    return {
      ngModule: SharedModule,
      providers: [],
    };
  }
}
