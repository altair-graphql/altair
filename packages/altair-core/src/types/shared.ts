export type IDictionary<V = any> = Record<string, V>;

export type TODO = any;

// export type Omit<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>>;

export interface BaseDocument {
  id: string;
  created_at?: number;
  updated_at?: number;
}

export interface BaseOwnableDocument extends BaseDocument {
  ownerUid: string;
}

type TypedOmit<T, K extends keyof T> = Omit<T, K>;
export type CreateDTO<
  T extends BaseDocument | BaseOwnableDocument
> = T extends BaseOwnableDocument
  ? TypedOmit<T, keyof BaseOwnableDocument>
  : TypedOmit<T, keyof BaseDocument>;
export type UpdateDTO<T extends BaseDocument | BaseOwnableDocument> = CreateDTO<
  T
> &
  Pick<T, 'id'>;
