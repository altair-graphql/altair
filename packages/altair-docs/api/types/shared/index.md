# types/shared

## Index

| Member | Description |
| :------ | :------ |
| [BaseDocument](interfaces/BaseDocument.md) | - |
| [BaseOwnableDocument](interfaces/BaseOwnableDocument.md) | - |
| [CreateDTO](type-aliases/CreateDTO.md) | - |
| [IDictionary](type-aliases/IDictionary.md) | - |
| [NoExtraProperties](type-aliases/NoExtraProperties.md) | - |
| [TODO](type-aliases/TODO.md) | - |
| [UpdateDTO](type-aliases/UpdateDTO.md) | - |
