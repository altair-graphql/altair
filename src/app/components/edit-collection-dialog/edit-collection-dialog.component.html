<nz-modal
  [nzVisible]="showEditCollectionDialog"
  (nzVisibleChange)="toggleDialogChange.next($event)"
  [nzTitle]="modalTitle"
  [nzContent]="modalContent"
  [nzFooter]="modalFooter"
  (nzOnCancel)="toggleDialogChange.next($event)"
>
  <ng-template #modalTitle>
    <div class="app-dialog-header">
      <div class="app-dialog-title">{{ 'EDIT_COLLECTION_TITLE_TEXT' | translate }}</div>
      {{ 'EDIT_COLLECTION_TITLE_SUBTEXT' | translate }}
    </div>
  </ng-template>

  <ng-template #modalContent>
    <div class="app-dialog-body" *ngIf="collection">
      <div class="form-group">
        {{ 'COLLECTION_TITLE_LABEL' | translate }}
        <input type="text" class="dialog-input" [(ngModel)]="collection.title">
      </div>
    </div>
  </ng-template>

  <ng-template #modalFooter>
    <div class="app-dialog-footer">
      <button class="app-button active-primary right" track-id="" (click)="updateCollection()">{{ 'SAVE_BUTTON' | translate }}</button>
    </div>
  </ng-template>
</nz-modal>
