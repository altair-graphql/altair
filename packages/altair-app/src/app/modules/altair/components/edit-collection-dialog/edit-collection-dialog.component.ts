import {
  Component,
  Input,
  Output,
  EventEmitter,
  OnChanges,
  SimpleChanges,
} from '@angular/core';
import { IQueryCollection } from 'altair-graphql-core/build/types/state/collection.interfaces';
import { PostrequestState } from 'altair-graphql-core/build/types/state/postrequest.interfaces';
import { PrerequestState } from 'altair-graphql-core/build/types/state/prerequest.interfaces';

@Component({
  selector: 'app-edit-collection-dialog',
  templateUrl: './edit-collection-dialog.component.html',
  styleUrls: ['./edit-collection-dialog.component.scss'],
})
export class EditCollectionDialogComponent implements OnChanges {
  @Input() showEditCollectionDialog = true;
  @Input() collection?: IQueryCollection;
  @Output() toggleDialogChange = new EventEmitter();
  @Output() importCurlChange = new EventEmitter<string>();
  @Output() updateCollectionChange = new EventEmitter<{
    collection: IQueryCollection;
  }>();

  title = '';
  preRequest: PrerequestState = { script: '', enabled: false };
  postRequest: PostrequestState = { script: '', enabled: false };

  ngOnChanges(changes: SimpleChanges) {
    if (changes.collection?.currentValue) {
      const collection = changes.collection?.currentValue as IQueryCollection;
      // setup form data fields
      this.title = collection.title;
      this.preRequest = collection.preRequest || { script: '', enabled: false };
      this.postRequest = collection.postRequest || {
        script: '',
        enabled: false,
      };
    }
  }

  updateCollection() {
    if (!this.collection) {
      throw new Error('this should never happen');
    }
    const collection = {
      ...this.collection,
      title: this.title,
      preRequest: this.preRequest,
      postRequest: this.postRequest,
    };
    this.toggleDialogChange.next(false);
    this.updateCollectionChange.next({ collection: collection });
  }
}
