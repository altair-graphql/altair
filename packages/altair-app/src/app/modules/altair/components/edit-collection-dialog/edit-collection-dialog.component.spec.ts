import { EditCollectionDialogComponent } from './edit-collection-dialog.component';
import { FormsModule } from '@angular/forms';
import {
  BrowserAnimationsModule,
  NoopAnimationsModule,
} from '@angular/platform-browser/animations';
import { SharedModule } from '../../modules/shared/shared.module';
import { mount, NgxTestWrapper } from '../../../../../testing';
import { NO_ERRORS_SCHEMA } from '@angular/core';
import { expect, it } from '@jest/globals';

describe('EditCollectionDialogComponent', () => {
  let wrapper: NgxTestWrapper<EditCollectionDialogComponent>;

  beforeEach(async () => {
    wrapper = await mount({
      component: EditCollectionDialogComponent,
      declarations: [EditCollectionDialogComponent],
      providers: [],
      imports: [NoopAnimationsModule, FormsModule, SharedModule],
      schemas: [NO_ERRORS_SCHEMA],
    });
  });

  it('should render correctly', () => {
    expect(wrapper.component.nativeElement).toMatchSnapshot();
  });
});
