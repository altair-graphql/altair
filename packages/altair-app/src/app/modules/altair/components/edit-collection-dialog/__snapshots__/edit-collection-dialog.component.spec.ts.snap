// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditCollectionDialogComponent should render correctly 1`] = `
<app-edit-collection-dialog>
  <nz-modal />
  
</app-edit-collection-dialog>
`;
